actors/ethaccount/src/lib.rs

```diff
@@ -46,67 +55,90 @@ impl EthAccountActor {
 
         Ok(())
     }
-}
 
-impl ActorCode for EthAccountActor {
-    type Methods = Method;
-    actor_dispatch! {
-        Constructor => constructor,
-    }
-}
+    /// Authenticates whether the provided signature is valid for the provided message.
+    /// Should be called with the raw bytes of a signature, NOT a serialized Signature object that includes a SignatureType.
+    /// Errors with USR_ILLEGAL_ARGUMENT if the authentication is invalid.
+    pub fn authenticate_message(
```

Comment:
The implementation of AuthenticateMessage is uglier than I'd like. I'm not sure whether some of the logic here already exists and I just didn't find it. If it doesn't we might want to move some of this to a shared place?

Reply:
See https://github.com/filecoin-project/ref-fvm/issues/974. But that's likely a larger refactor involving the EVM/EAM/EthAccount.